hibernate-core/src/main/java/org/hibernate/hql/internal/ast/util/JoinProcessor.java

```diff
@@ -48,6 +59,9 @@
 public class JoinProcessor implements SqlTokenTypes {
 	private static final CoreMessageLogger LOG = CoreLogging.messageLogger( JoinProcessor.class );
 
+	// TODO where to place the default value?
+	public static final String DEFAULT_OMIT_JOIN_OF_SUPERCLASS_TABLES = Boolean.TRUE.toString();
```

Comment:
Please add the configuration option to `org.hibernate.boot.spi.SessionFactoryOptions`. Take a look at how `isCollectionJoinSubqueryRewriteEnabled()` is implemented and initialized. There you can see how to set the default value.